### ISSUE TITLE
CSLA breaks other authorization policies 

### ISSUE THREAD

**Issue by j055** [MarimerLLC/csla#2000]
**Describe the bug**
If I add a policy in `Startup.ConfigureServices` with `services.AddCsla().WithBlazorServerSupport()` I will get an error `The AuthorizationPolicy named: 'TokenAuthentication' was not found.` when calling an MVC action which references my policy.

**Version and Platform**
CSLA version: 5.4.0

**Code that Fails**

Add a policy:
```            
services.AddAuthorization(options => options.AddPolicy(Constants.TokenAuthentication,
                          policy => policy.AddAuthenticationSchemes(Constants.TokenAuthentication).RequireAuthenticatedUser()));
```

and an attribute to an MVC controller:

`  [Authorize(Policy = Constants.TokenAuthentication)]`

`GetPolicyAsync` seems to ignore my custom policy and returns the fallback policy instead

```
    public Task<AuthorizationPolicy> GetPolicyAsync(string policyName)
    {
      CslaPermissionRequirement requirement;
      if (!CslaPolicy.TryGetPermissionRequirement(policyName, out requirement))
        return this.GetFallbackPolicyAsync(); 
      AuthorizationPolicyBuilder authorizationPolicyBuilder = new AuthorizationPolicyBuilder(Array.Empty<string>());
      authorizationPolicyBuilder.AddRequirements((IAuthorizationRequirement) requirement);
      return Task.FromResult<AuthorizationPolicy>(authorizationPolicyBuilder.Build());
    }
```

Remove this line and my policy works:

```
//services.AddCsla().WithBlazorServerSupport();
```

**Stack Trace or Exception Detail**
System.InvalidOperationException: The AuthorizationPolicy named: 'TokenAuthentication' was not found.

**Comment by rockfordlhotka** [comment]
I debated whether to include the policy code in the `WithBlazorServerSupport` method.

Maybe the solution is that I have a bug in my `GetPolicyAsync` method. Maybe the solution is to force you to manually write that config code in your startup.

**Comment by j055** [comment]
Can you not add the CSLA policies to the builtin `AuthorizationPolicyProvider` and put the implementation
 in the `WithBlazorServerSupport` extension method?

e.g.
```
services.AddAuthorization(options => options.AddPolicy("CSLA:",
                          policy => policy.AddRequirements(/*add requirements*/)));
``` 
https://github.com/dotnet/aspnetcore/blob/master/src/Security/Authorization/Core/src/DefaultAuthorizationPolicyProvider.cs

Looks like `GetPolicyAsync` should just be getting the policy from an existing dictionary. I'm not sure you even need a `CslaPermissionsPolicyProvider`. Doesn't the builtin implementation give you what you need?

I do hope I haven't missed the point entirely but please let me know if I'm on to something.

**Comment by j055** [comment]
Your `ICslaBuilder` might need to contain the `IServiceCollection` or `AddCsla` can have an optional `System.Action` for `WithBlazorServerSupport` or something else better that this.

**Comment by rockfordlhotka** [comment]
The reason the authz stuff is added is that CSLA actually requires it be configured, and authz is not configured by default. I don't think it matters if it is _csla_ authz that's configured, but _some authz_ must be configured.

**Comment by j055** [comment]
I think we should be checking any existing policies then as per my pull request. What do you think?

https://github.com/MarimerLLC/csla/pull/2005

**Comment by rockfordlhotka** [comment]
@j055 just to be clear - the PR was into `master`, so CSLA 6. If you'd like to fix it in 5.4.x I'll accept that PR too.
